### PR TITLE
Docker: minor fixes to build and push automation

### LINKED ISSUE #450: Docker: fixes to automatic build and push
- build and push on new tag not new release
- git clone with depth=1 for lighter resulting image
- git clone version associated to tag
- docker tag and push under both tagged version and "latest"

### PR DESCRIPTION
**Pull request by remtav**
fixes #450 

Github workflow .yml:
- build and push on new tag not new release
- docker tag and push under both tagged version and "latest" Dockerfile:

Dockerfile:
- use single-stage build, not multi-stage to reduce disk usage during docker build (github's workflow was running out of space)
- git clone with depth=1 for lighter resulting image
- git clone version associated to tag with GIT_TAG arg
- restore option to use certificate if building behind firewall

### CODE REVIEW

**Review by ms888ekb — APPROVED** [review]

**Review by CharlesAuthier — DISMISSED** [review]

File: .github/workflows/build-push-on-release.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-on:
-  release:
-    types: [published]
-
+on: push
+#  push:
+#    tags:
+#      - v**
```
Comment (remtav) [inline .github/workflows/build-push-on-release.yml:4]: just for testing

**Comment by ms888ekb** [comment]
I remember you mentioned that the resulting SIF image didn't have a gpu support lest time you switched to mamba. Does it work now?

**Review by valhassan — APPROVED** [review]

**Comment by remtav** [comment]
> * estore option to use certificate if building behind firewall

Yup, there's still a bug with using mamba to create environnement while still getting pytorch-gpu. I've dropped using mamba for now. We still use conda in this PR: https://github.com/NRCan/geo-deep-learning/pull/451/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R41

**Comment by remtav** [comment]
> I remember you mentioned that the resulting SIF image didn't have a gpu support lest time you switched to mamba. Does it work now?

@ms888ekb Yup, there's still a bug with using mamba to create environnement while still getting pytorch-gpu. I've dropped using mamba for now. We still use conda in this PR: https://github.com/NRCan/geo-deep-learning/pull/451/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R41

**Review by ms888ekb — APPROVED** [review]

**Pull request merged**